MEDMI_Data_Download/met_extraction_script.py

```diff
@@ -65,37 +80,114 @@ def extraction_wind_function(source_dict,settings_dict):
+	global VERBOSE
+	DEFAULT_OUT_DIR_PREFIX = 'data_'
+	DEFAULT_DATE_RANGE = ['2016-1-1 0','2019-12-31 23']
+	DEFAULT_DATES_STRING = '2016-2019'
+	DEFAULT_LATITUDES = [48,60]
+	DEFAULT_LONGITUDES = [-11,3]
+	DEFAULT_VERBOSE = 0
+
+	### general help text
+	parser = argparse.ArgumentParser(
+		description="*** A script for automated downloading of MET data for a given date range. ***")
+
+	### parameters
+
+	# filenames
+	parser.add_argument("--outdir_prefix", "-o", dest="outdir_prefix",
+						help="prefix to be appended on output directory name. Default: " + DEFAULT_OUT_DIR_PREFIX)
+	parser.set_defaults(output_prefix=DEFAULT_OUT_DIR_PREFIX)
+
+	# Dates
+	parser.add_argument("--date_range", "-d", dest="date_range", type=str, nargs='+',  help="start and end dates \
+	 					(array - first two values only). Default: " + str(DEFAULT_DATE_RANGE))
+	parser.add_argument("--dates_string", "-s", dest="dates_string", help= "date string to be used in file search. \
+																		   Default: " + DEFAULT_DATES_STRING)
+
+	# Latitude / longitude
+	parser.add_argument("--latitude_range", "-t", dest="latitude_range", type=int, nargs='+',
+						help="start and end latitude range (array - first two values only). Default: " + str(DEFAULT_LATITUDES))
+	parser.add_argument("--longitude_range", "-n", dest="longitude_range", type=int, nargs='+',
+						help="start and end longitude range (array - first two values only). Default: " + str(DEFAULT_LONGITUDES))
+
+	# Log verbose-ness
+	parser.add_argument("--verbose", "-v", type=int,
+						help="Level of output for debugging (Default: " + str(DEFAULT_VERBOSE) + " (0 = verbose output))")
+
+	### Process Inputs
+	args = parser.parse_args()
+
+	if args.outdir_prefix:
+		outdir_prefix = args.outdir_prefix
+	else:
+		print('No outdir_prefix given, so will use default: ' + DEFAULT_OUT_DIR_PREFIX)
+		outdir_prefix = DEFAULT_OUT_DIR_PREFIX
+
+	if args.date_range:
+		date_range = args.date_range
+	else:
+		print('No date_range provided, so using default: ' + str(DEFAULT_DATE_RANGE))
+		date_range = DEFAULT_DATE_RANGE
+
+	if args.dates_string:
+		dates_string = args.dates_string
+	else:
+		print('No dates_string provided, so using default: ' + str(DEFAULT_DATES_STRING))
+		dates_string = DEFAULT_DATES_STRING
+
+	if args.latitude_range:
+		latitude_range = args.latitude_range
+	else:
+		print('No latitude_range provided, so using default: ' + str(DEFAULT_LATITUDES))
+		latitude_range = DEFAULT_LATITUDES
+
+	if args.longitude_range:
+		longitude_range = args.longitude_range
+	else:
+		print('No longitude_range provided, so using default: ' + str(DEFAULT_LONGITUDES))
+		longitude_range = DEFAULT_LONGITUDES
+
+	if args.verbose:
+		VERBOSE = max(args.verbose, 0)
+		print('verbose: ', VERBOSE)
+	else:
+		print('No verbose flag provided, so using default: ' + str(DEFAULT_VERBOSE))
+		VERBOSE = DEFAULT_VERBOSE
+
+
+
+	dict_base = {'Time range':date_range,'Latitude range': [48,60], 'Longitude range': [-11,3]}
```

Comment:
do these need to be replaced with the latitude and longitude ranges given above?

Reply:
yes, have done this.

Reply:
I have added checks to all the range inputs: long, lat and date to check at least 2 values and raise error if not with useful message. 
I've made use of  str.format()  throughout, instead of using + (and worst of all I was combining the two in the same line :/ )
I've also changed the dame of dates_string, to outfile_suffix, as that is essentially what it is.  It should be up to the user what to call their files, and they might want to add info about lat/long ranges and other inputs, as well as date range. 
... + they also have the option of adding a prefix to the directory name, to help with the organisation of outputs.